EJB webservice with wssecurity policy gets NoClassDefFoundError: org/apache/commons/logging/LogFactory

We have an ejb soap webservice with a ws security policy. We get a NoClassDefFoundError when receiving a message.

    [..][Payara 4.1][SEVERE][][com.sun.xml.ws.transport.http.HttpAdapter][tid: _ThreadID=40 _ThreadName=http-listener-2(3)][timeMillis:...][levelValue:1000][[
    org/apache/commons/logging/LogFactory
    java.lang.NoClassDefFoundError: org/apache/commons/logging/LogFactory
    at org.apache.xml.security.Init.<clinit>(Init.java:59)
    at com.sun.xml.ws.security.opt.impl.incoming.SecurityRecipient.<clint>(SecurityRecipient.java:132)
    at com.sun.xml.wss.provider.wsit.WSITServerAuthContext.verifyInboundMessage(WSITServerAuthContext.java:587)
    at com.sun.xml.wss.provider.wsit.WSITServerAuthContext.validateRequest(WSITServerAuthContext.java:364)
    at com.sun.xml.wss.provider.wsit.WSITServerAuthContext.validateRequest(WSITServerAuthContext.java:267)
    at com.sun.enterprise.security.webservices.CommonServerSecurityPipe.process(CommonServerSecurityPipe.java:141)
    at com.sun.xml.ws.api.pipe.helper.PipeAdapter.processRequest(PipeAdapter.java:119)
    at com.sun.xml.ws.api.pipe.Fiber.__doRun(Fiber.java:1136)
    at com.sun.xml.ws.api.pipe.Fiber._doRun(Fiber.java:1050)
    at com.sun.xml.ws.api.pipe.Fiber.doRun(Fiber.java:1019)
    at com.sun.xml.ws.api.pipe.Fiber.runSync(Fiber.java:877)
    at (WSEndpointImpl.java:419)
    at (HttpAdapter.java:868)
    at (HttpAdapter.java:422)
    at (ServletAdapter.java:169)
    at (Ejb3MessageDispatcher.java:110)
    at (Ejb3MessageDispatcher.java:80)
    at (EjbWebServiceServlet.java:210)
    at (EjbWebServiceServlet.java:153)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
    at org.glassfish.grizzly.servlet.ServletHandler.doServletService(ServletHandler.java:227)
    at (ServletHandler:179)
    at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
    at (ContainerMapper.java:167)
    at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:204)
    at (HttpHandler.java:178)
    at (HttpServerFilter.java:235)
    at (ExecutorResolver.java:119)
    at (DefaultFilterChange.java:283)
    at (DefaultFilterChange.java:200)
    at (DefaultFilterChange.java:131)
    at (DefaultFilterChange.java:111)
    at (ProcessExecutor.java:77)
    ...
    at java.lang.Thread.run(Thread.java:745)
    Caused by: java.lang.ClassNotFoundException: org.apache.commons.logging.LogFactory not found by org.glassfish.metro.webservices-osgi [217]
       at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1532)
       at org.apache.felix.framework.BundleWiringImpl.access$400(BundleWiringImpl.java:75)
       at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:1955)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
       ... 42 more
    ]]

--- a/appserver/pom.xml
+++ b/appserver/pom.xml
@@ -108,7 +108,7 @@
         <jaxb.version>2.2.12-b141219.1637</jaxb.version>
         <stax2-api.version>3.1.1</stax2-api.version>
         <javax.xml.registry-api.version>1.0.5</javax.xml.registry-api.version>
-        <eclipselink.version>2.5.2.p1</eclipselink.version>
+        <eclipselink.version>2.6.0-RC1</eclipselink.version>
         <javax-persistence-api.version>2.1.0</javax-persistence-api.version> 
         <dbschema.version>3.1.1</dbschema.version>
         <dbschema.osgi.version>6.0</dbschema.osgi.version>

--- a/appserver/pom.xml
+++ b/appserver/pom.xml
@@ -127,8 +127,8 @@
         <concurrent-api.version>1.0</concurrent-api.version>
         <concurrent.version>1.0</concurrent.version>
         <javax.batch-api.version>1.0.1-b01</javax.batch-api.version>
-        <com.ibm.jbatch.container.version>1.0.1-b04</com.ibm.jbatch.container.version>
-        <com.ibm.jbatch.spi.version>1.0.1-b04</com.ibm.jbatch.spi.version>
+        <com.ibm.jbatch.container.version>1.0.1-b07</com.ibm.jbatch.container.version>
+        <com.ibm.jbatch.spi.version>1.0.1-b07</com.ibm.jbatch.spi.version>
         <javax.xml.soap-api.version>1.3.7</javax.xml.soap-api.version>
 	<javax.management.j2ee-api.version>1.1.1</javax.management.j2ee-api.version>
         <jboss.logging.version>3.1.3.GA</jboss.logging.version>

--- a/appserver/pom.xml
+++ b/appserver/pom.xml
@@ -101,14 +101,14 @@
         <jsftemplating.version>2.1.1</jsftemplating.version>
         <uc-pkg-client.version>1.122-57.2889</uc-pkg-client.version>
         <uc-pkg-bootstrap.version>1.122-57.2889</uc-pkg-bootstrap.version>
-        <webservices.version>2.3.2-b608</webservices.version>
+        <webservices.version>2.3.1</webservices.version>
         <jsr181-api.version>1.0-MR1</jsr181-api.version>
         <woodstox.version>4.2.0</woodstox.version>
-        <jaxb-api.version>2.2.13-b141020.1521</jaxb-api.version>
-        <jaxb.version>2.2.12-b141219.1637</jaxb.version>
+        <jaxb-api.version>2.2.12-b140109.1041</jaxb-api.version>
+        <jaxb.version>2.2.10-b140802.1033</jaxb.version>
         <stax2-api.version>3.1.1</stax2-api.version>
         <javax.xml.registry-api.version>1.0.5</javax.xml.registry-api.version>
-        <eclipselink.version>2.6.0-RC1</eclipselink.version>
+        <eclipselink.version>2.5.2.p1</eclipselink.version>
         <javax-persistence-api.version>2.1.0</javax-persistence-api.version> 
         <dbschema.version>3.1.1</dbschema.version>
         <dbschema.osgi.version>6.0</dbschema.osgi.version>
@@ -127,8 +127,8 @@
         <concurrent-api.version>1.0</concurrent-api.version>
         <concurrent.version>1.0</concurrent.version>
         <javax.batch-api.version>1.0.1-b01</javax.batch-api.version>
-        <com.ibm.jbatch.container.version>1.0.1-b07</com.ibm.jbatch.container.version>
-        <com.ibm.jbatch.spi.version>1.0.1-b07</com.ibm.jbatch.spi.version>
+        <com.ibm.jbatch.container.version>1.0.1-b04</com.ibm.jbatch.container.version>
+        <com.ibm.jbatch.spi.version>1.0.1-b04</com.ibm.jbatch.spi.version>
         <javax.xml.soap-api.version>1.3.7</javax.xml.soap-api.version>
 	<javax.management.j2ee-api.version>1.1.1</javax.management.j2ee-api.version>
         <jboss.logging.version>3.1.3.GA</jboss.logging.version>

--- a/nucleus/pom.xml
+++ b/nucleus/pom.xml
@@ -106,6 +106,17 @@
 
     <repositories>
         <repository>
+            <id>jvnet-nexus-releases</id>
+            <name>Java.net Releases Repositories</name>
+            <url>https://maven.java.net/content/repositories/releases/</url>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+        </repository>
+        <repository>
             <id>jvnet-nexus-promoted</id>
             <name>Java.net Promoted Repositories</name>
             <url>https://maven.java.net/content/repositories/promoted/</url>

